File: JS/boj_10828_스택.js
const fs = require('fs');

// 공백으로 구분된 input값을 불러온다.
// const input = fs.readFileSync('/dev/stdin').toString().split('\r\n');
// 전개연산자를 사용하여 T와 괄호를 구분한다.
const [T, ...arr] = fs.readFileSync('10828.txt').toString().split('\r\n');
// console.log(T)
// console.log(arr)

const stack = []
for (var i = 0; i < parseInt(T); i++) {
  const order = arr[i].split(' ')
  // console.log(order)
  if (order[0] === 'push') {
    stack.push(order[1])
  } else if (order[0] === 'pop') {
    if (stack.length > 0) {
      console.log(parseInt(stack[stack.length-1]))
      stack.pop()
    } else{
      console.log(-1)
    }
  } else if (order[0] === 'size') {
    console.log(parseInt(stack.length))
  } else if (order[0] === 'empty') {
    if (stack.length > 0) {
      console.log(0)
    } else {
      console.log(1)
    }
  } else if (order[0] === 'top') {
    if (stack.length > 0) {
      console.log(parseInt(stack[stack.length-1]))
    } else {
      console.log(-1)
    }
  }
}

File: boj/boj_10828_스택.py
import sys; sys.stdin = open("input/10828.txt", "r")

N = int(sys.stdin.readline().rstrip())
arr = list(sys.stdin.readline().rstrip().split(' ') for _ in range(N))
stack = []

for i in range(N):
    if arr[i][0] == 'push':
        stack.append(arr[i][1])
    elif arr[i][0] == 'pop':
        if len(stack) > 0:
            print(stack.pop())
        else:
            print(-1)
    elif arr[i][0] == 'size':
        print(len(stack))
    elif arr[i][0] == 'empty':
        if len(stack) > 0:
            print(0)
        else:
            print(1)
    else:
        if len(stack) > 0:
            print(stack[-1])
        else:
            print(-1)
